pkgs/applications/editors/neovim/default.nix

```diff
@@ -15,22 +16,32 @@
 }:
 
 let
-  neovimLuaEnv = lua.withPackages(ps:
-    (with ps; [ lpeg luabitop mpack ]
-    ++ lib.optionals doCheck [
-        nvim-client luv coxpcall busted luafilesystem penlight inspect
-      ]
-    ));
+  requiredLuaPkgs = ps: (with ps; [
+    lpeg
+    luabitop
+    mpack
+  ] ++ lib.optionals doCheck [
+    nvim-client
+    luv
+    coxpcall
+    busted
+    luafilesystem
+    penlight
+    inspect
+  ]
+  );
+  neovimLuaEnv = lua.withPackages requiredLuaPkgs;
+  neovimLuaEnvFB = lua.luaOnBuild.withPackages requiredLuaPkgs;
   codegenLua =
-    if lua.pkgs.isLuaJIT
+    if lua.luaOnBuild.pkgs.isLuaJIT
```

Comment:
is luaOnBuild a common pattern. I found a similar pattern for python, have you just copied that ?

Reply:
I copied the pattern from python when i fixed splicing for `lua.pkgs`
https://github.com/nixos/nixpkgs/commit/39571bd6fa6d558c45d67e91333a3cfda645b738

xOnBuild is a useful pattern for packages which have a pkgs because if someone overrides `lua` the `luaOnBuild` will be overriden too